### PR TITLE
#164139690 Add feature to follow and unfollow users

### PR DESCRIPTION
**Pull request by youngestdj**
#### What does this PR do? 
This PR adds a feature to follow and unfollow users
#### Description of task to be completed?
* Add table `follows`
* Add endpoint to follow users
* Add endpoint to unfollow users
#### How should this be manually tested?
* Run `npm run db-migrate` to run migrations
* Run `npm start` to start the application
* Create a two users by sending two post requests to /api/v1/user
* Verify the users by clicking the link in your email
* Send a `GET` request along with the user id to` /api/v1/followuser/<id>` 
* To unfollow the user send a `GET` request along with the user id to` /api/v1/unfollowuser/<id>`
#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164139690

### CODE REVIEW

File: models/follows.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module.exports = (sequelize, DataTypes) => {
+  const follows = sequelize.define('follows', {
+    followerId: DataTypes.INTEGER,
+    followingId: DataTypes.INTEGER
+  }, {});
+  follows.associate = () => {
```
Comment (kleva-j) [inline models/follows.js]: I think you should take out this association function since you are not using it.

**Review by kleva-j — APPROVED** [review]

**Review by kleva-j — APPROVED** [review]

**Review by olajide1234 — APPROVED** [review]

**Review by codinger41 — APPROVED** [review]

File: doc.json [hunk 2]
```diff
@@ -342,6 +342,88 @@
...
+          "200": {
+            "description": "User followed successfully."
+          },
+          "403": {
+            "description": "You cannot follow a user twice."
+          }
+        }
+      }
+    },
+    "/unfollowuser/{id}": {
```
Comment (9jaswag) [inline doc.json]: @youngestdj `follow` and `unfollow` should suffice as route names except we plan to implement some other `follow` feature in the future.

File: controllers/follow.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  /**
+   * @description follow a user
+   * @param {object} req http request object
+   * @param {object} res http response object
+   * @returns  {object} response
+   */
+  static async followUser(req, res) {
+    const { userFollowing, userToFollow, userToFollowId } = req;
+    // Check if user is already following the user and return an error if they are
+    const getFollowing1 = await userFollowing.getFollowing();
```
Comment (9jaswag) [inline controllers/follow.js]: @youngestdj why is this `getFollowing1`? Is there `getFollowing2`?

File: controllers/follow.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    let userIsFollowing = false;
+    getFollowing1.forEach((element) => {
```
Comment (9jaswag) [inline controllers/follow.js]: @youngestdj is there a better way to do this? If the user has 1 million followers, it means we're gonna loop through them all just to find if one exists.
There's a way to do this but I'm not sure if Sequelize offers that. See if you can get an array of `ids` for all the followers via a Sequelize method, then you can just do an `includes` to find if the id exists. cc @leksyib you can check if this exists in Sequelize too.

File: controllers/follow.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+   */
+  static async unfollowUser(req, res) {
+    const { userFollowing } = req;
+    const userToUnfollow = req.userToFollow;
+    const userToUnfollowId = req.userToFollowId;
+
+    // Check if user is not following the user and return an error if they aren't
+    const getFollowing1 = await userFollowing.getFollowing();
+    let userIsFollowing = false;
+    getFollowing1.forEach((element) => {
```
Comment (9jaswag) [inline controllers/follow.js]: @youngestdj same thing here

File: controllers/follow.js (hunk 4 above)
Comment (codinger41) [reply controllers/follow.js]: Alright @9jaswag

File: doc.json (hunk 2 above)
Comment (youngestdj) [reply doc.json]: Route names have been changed

File: controllers/follow.js (hunk 3 above)
Comment (youngestdj) [reply controllers/follow.js]: This has been removed

File: controllers/follow.js (hunk 4 above)
Comment (youngestdj) [reply controllers/follow.js]: I have used an inbuilt Sequelize function to achieve this

File: controllers/follow.js (hunk 5 above)
Comment (youngestdj) [reply controllers/follow.js]: This has been removed also

**Review by 9jaswag — APPROVED** [review]

**Pull request merged**